[CI] TSDBSyntheticIdPostingsFormatTests testSeek failing

**Build Scans:**
- [elasticsearch-periodic-platform-support #12355 / oraclelinux-9_platform-support-unix](https://gradle-enterprise.elastic.co/s/zgdbbqvfbj3ua)
- [elasticsearch-periodic-platform-support #12355 / oraclelinux-9_platform-support-unix](https://gradle-enterprise.elastic.co/s/gudi2hlkj52hc)
- [elasticsearch-periodic-platform-support #12329 / debian-12_platform-support-unix](https://gradle-enterprise.elastic.co/s/h2bji32yxzjhe)
- [elasticsearch-periodic-platform-support #12329 / debian-12_platform-support-unix](https://gradle-enterprise.elastic.co/s/wadnksqbd6aa4)

**Reproduction Line:**
```
./gradlew ":server:test" --tests "org.elasticsearch.index.codec.tsdb.TSDBSyntheticIdPostingsFormatTests.testSeek" -Dtests.seed=4119B3F1B8CF217A -Dtests.locale=nso-ZA -Dtests.timezone=America/North_Dakota/New_Salem -Druntime.java=26
```

**Applicable branches:**
9.4

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.index.codec.tsdb.TSDBSyntheticIdPostingsFormatTests),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!('testSeek'),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: 
Expected: <END>
     but: was <NOT_FOUND>
```

**Issue Reasons:**
- [9.4] 2 consecutive failures in step oraclelinux-9_platform-support-unix
- [9.4] 4 failures in test testSeek (1.5% fail rate in 269 executions)
- [9.4] 2 failures in step debian-12_platform-support-unix (25.0% fail rate in 8 executions)
- [9.4] 2 failures in step oraclelinux-9_platform-support-unix (28.6% fail rate in 7 executions)
- [9.4] 2 failures in pipeline elasticsearch-periodic-platform-support (28.6% fail rate in 7 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -497,9 +497,6 @@ tests:
 - class: org.elasticsearch.xpack.stateless.snapshots.StatelessSnapshotStressTestsIT
   method: testRandomActivitiesStatelessSnapshotDisabled
   issue: https://github.com/elastic/elasticsearch/issues/150884
-- class: org.elasticsearch.index.codec.tsdb.TSDBSyntheticIdPostingsFormatTests
-  method: testSeek
-  issue: https://github.com/elastic/elasticsearch/issues/150912
 - class: org.elasticsearch.xpack.stateless.StatelessMergeIT
   method: testOnlyForceMergesAreAllowedDuringRelocation
   issue: https://github.com/elastic/elasticsearch/issues/150925

--- a/server/src/test/java/org/elasticsearch/index/codec/tsdb/TSDBSyntheticIdPostingsFormatTests.java
+++ b/server/src/test/java/org/elasticsearch/index/codec/tsdb/TSDBSyntheticIdPostingsFormatTests.java
@@ -912,7 +912,7 @@ public class TSDBSyntheticIdPostingsFormatTests extends ESTestCase {
             modifiedTimestamp = timestamp;
         }
 
-        final var term = TsidExtractingIdFieldMapper.createSyntheticIdBytesRef(
+        final var term = createSyntheticIdBytesRef(
             modifiedTsId,
             modifiedTimestamp,
             TsidExtractingIdFieldMapper.extractRoutingHashFromSyntheticId(value)
@@ -934,19 +934,19 @@ public class TSDBSyntheticIdPostingsFormatTests extends ESTestCase {
             // tsids are not identical, find the first byte that differs
             int diffIndex = 0;
             final int minLen = Math.min(tsIdMin.length, tsIdMax.length);
-            while (diffIndex < minLen && min.bytes[min.offset + diffIndex] == max.bytes[max.offset + diffIndex]) {
+            while (diffIndex < minLen && tsIdMin.bytes[tsIdMin.offset + diffIndex] == tsIdMax.bytes[tsIdMax.offset + diffIndex]) {
                 diffIndex++;
             }
 
             // increment the first byte that differs (if there is room for doing so)
             if (diffIndex < minLen) {
-                int valueMin = min.bytes[min.offset + diffIndex] & 0xFF;
-                int valueMax = (diffIndex < max.length) ? (max.bytes[max.offset + diffIndex] & 0xFF) : 256;
+                int valueMin = tsIdMin.bytes[tsIdMin.offset + diffIndex] & 0xFF;
+                int valueMax = (diffIndex < tsIdMax.length) ? (tsIdMax.bytes[tsIdMax.offset + diffIndex] & 0xFF) : 256;
                 if (valueMax - valueMin > 1) {
                     byte[] tsid = new byte[tsIdMin.length];
                     System.arraycopy(tsIdMin.bytes, tsIdMin.offset, tsid, 0, tsIdMin.length);
                     tsid[diffIndex] = (byte) (valueMin + randomIntBetween(1, valueMax - valueMin - 1));
-                    return TsidExtractingIdFieldMapper.createSyntheticIdBytesRef(
+                    return createSyntheticIdBytesRef(
                         new BytesRef(tsid),
                         TsidExtractingIdFieldMapper.extractTimestampFromSyntheticId(min),
                         TsidExtractingIdFieldMapper.extractRoutingHashFromSyntheticId(min)
@@ -961,13 +961,13 @@ public class TSDBSyntheticIdPostingsFormatTests extends ESTestCase {
         long diffTimestamps = Math.abs(timestampMin - timestampMax);
         if (diffTimestamps > 1L) {
             if (timestampMin > timestampMax) {
-                return TsidExtractingIdFieldMapper.createSyntheticIdBytesRef(
+                return createSyntheticIdBytesRef(
                     tsIdMin,
                     timestampMin - randomLongBetween(1L, diffTimestamps - 1L),
                     TsidExtractingIdFieldMapper.extractRoutingHashFromSyntheticId(min)
                 );
             } else {
-                return TsidExtractingIdFieldMapper.createSyntheticIdBytesRef(
+                return createSyntheticIdBytesRef(
                     tsIdMax,
                     timestampMax + randomLongBetween(1L, diffTimestamps - 1L),
                     TsidExtractingIdFieldMapper.extractRoutingHashFromSyntheticId(max)
@@ -975,6 +975,17 @@ public class TSDBSyntheticIdPostingsFormatTests extends ESTestCase {
             }
         }
         return null; // Nothing we can do, min and max have identical _tsid and @timestamp
+    }
+
+    static BytesRef createSyntheticIdBytesRef(BytesRef tsid, long timestamp, int routingHash) {
+        BytesRef id = TsidExtractingIdFieldMapper.createSyntheticIdBytesRef(tsid, timestamp, routingHash);
+        if (Byte.toUnsignedInt(id.bytes[id.offset]) >= Uid.BASE64_ESCAPE) {
+            byte[] newBytes = new byte[id.length + 1];
+            System.arraycopy(id.bytes, id.offset, newBytes, 1, id.length);
+            newBytes[0] = (byte) Uid.BASE64_ESCAPE;
+            return new BytesRef(newBytes);
+        }
+        return id;
     }
 
 }

